python-mini-projects/projects/Random_word_from_list/Random_word_from_list.py

```diff
@@ -0,0 +1,29 @@
+import sys
+import random
+
+# check if filename is supplied as a command line argument
+if sys.argv[1:]:
+    filename = sys.argv[1]
+else:
+    filename = input("What is the name of the file? (extension included): ")
+
+try:
+    file = open(filename)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *file*. The resource is allocated by call *builtins.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *file* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).